flow/templates/custom.sh

```diff
@@ -0,0 +1,9 @@
+{{ "{{% extends " + extend_template + " %}}" }}
```

Comment:
I would recommend a more descriptive file name, like `custom_script_template.sh`. Maybe we should change the terminology of this PR to call it a "user environment template"?

This needs a nice comment block to explain what it's doing.
```suggestion
{#
This is a jinja template used to generate another jinja template.
This file is used by `flow template create` to detect and create a new script
template that can be modified by the user to add custom commands (e.g.
`module load ...`).
#}
{{ "{{% extends " + extend_template + " %}}" }}
```